build/builder.go

```diff
@@ -539,29 +583,100 @@ func (b *Builder) Finish() error {
 		return b.buildError
 	}
 
+	// map of temporary -> final names for all updated shards + shard metadata files
+	artifactPaths := make(map[string]string)
+	for tmp, final := range b.finishedShards {
+		artifactPaths[tmp] = final
+	}
+
+	oldShards := b.opts.FindAllShards()
+
+	if b.opts.IsDelta {
+		// Delta shard builds need to update FileTombstone and branch commit information for all
+		// existing shards
+		for _, shard := range oldShards {
+			repositories, indexMetadata, err := zoekt.ReadMetadataPathAlive(shard)
+			if err != nil {
+				b.buildError = err
+				continue
+			}
+
+			for _, r := range repositories {
+				if r.ID == b.opts.RepositoryDescription.ID {
+
+					if len(b.opts.ChangedOrRemovedFiles) > 0 && r.FileTombstones == nil {
```

Comment:
I think we only need to set tombstones if `len(b.opts.ChangeOrRemovedFiles)>0`. Maybe it would make sense to move this check to line 594 and we might be able to skip a lot of work if just files were added.

Reply:
This situation is actually tested in https://github.com/sourcegraph/zoekt/pull/274/files#diff-620acbfcb363a777e9bca5369361398fbc299ddaa61a1670402550375c914c59R426

Reply:
Oh yes, I forgot about the version metadata 🤦 .